src/main/java/Duke.java

```diff
@@ -1,10 +1,358 @@
+                storage.updateLocalFile(tasks.get());
+                taskCount = Task.getTaskCount();
+                return ui.deleteAnnouncement(taskDelete, taskCount);
+            } else if (!check.equals("bye")) {
+                return createTask(userCommand, due, taskDescription, storage, ui, tasks);
+            } else if (check.equals("bye")) {
+                return ui.sayYourGoodBye();
+            }
+        } catch (DukeException ex) {
+            System.out.println("OOPS!!! " + ex.getMessage() + "\n");
+        }
+
+        return "Are you speaking in botlang? I don't understand what you are saying!";
+    }
+
+    /**
+     * Converts user input in the form of String to the format of Date.
+     * The Task Manager has the ability to read dates instead of taking dates as String.
+     *
+     * @param input The date/ time input in the form of String. It should follow either dd/MM/yyyy HHmm or HHmm format.
+     * @return Date format of the input String.
+     */
+    private static Date convertStringToDate(String input) {
+        Date date = new Date();
+        try {
+            if (input.length() <= 5) {
+                SimpleDateFormat formatTimeOnly = new SimpleDateFormat("HHmm");
+                date = formatTimeOnly.parse(input.trim());
+                return date;
+            } else {
+                SimpleDateFormat format = new SimpleDateFormat("dd/MM/yyyy HHmm", Locale.ENGLISH);
+                date = format.parse(input);
+                return date;
+            }
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        return date;
+    }
+
+    /**
+     * Performs the operation of creating a Task (Todo, Deadline, Task).
+     * It also writes the list of tasks to the local storage.
+     *
+     * @param userCommand     Indicates the instruction (done, delete, etc) to the bot.
+     * @param due             Provides the due date/ time of the task.
+     * @param taskDescription Reflects the description of a task.
+     * @param storage         The local storage for the Tasks as a Storage object.
+     * @param ui              The User Interface object that performs all interactions with the user.
+     * @param tasks           The list of Task that are temporary stored with the bot.
+     * @return Gives a feedback to the user on the operation has performed after a command is given.
+     * @throws EmptyToDoDescriptionException Indicates an empty description that should not be left empty for Todo.
+     * @throws EmptyDescriptionException     Indicates an empty description that should not be left empty.
+     * @throws UnknownCommandException       Indicates the inability of the bot to read the command that is given by the user.
+     */
+    public static String createTask(String userCommand, String due, String taskDescription, Storage storage, Ui ui, TaskList tasks) throws EmptyToDoDescriptionException, EmptyDescriptionException, UnknownCommandException {
```

Comment:
Perhaps you can perform more OOP and break this method up. 6 arguments sounds a bit like overkill